### PR TITLE
Operational defensive baseline: strict test-user gate, avatar quota, Node engines

### PR DESCRIPTION
**Pull request by Meganeuridae**
Three small operational-safety items from the Tier 1 queue, separate commits.

## 1. `fix(security)`: require explicit env-var opt-in to create test users

The previous gate — `if (NODE_ENV !== 'production')` — fails open. An unset, empty, misspelled, or differently-cased `NODE_ENV` (`prod`, `Production`, ``) silently bypasses the check and creates test users with known credentials on what is effectively a prod host.

Replaces the negative check with an explicit allowlist:

```js
NODE_ENV === 'development' || NODE_ENV === 'test'
  || ENABLE_TEST_USERS === 'true'
```

Conventional local-dev (`NODE_ENV=development`) and conventional CI test runs (`NODE_ENV=test`) keep working unchanged. The `ENABLE_TEST_USERS` opt-in covers edge cases. Anything else — including every shape of misconfiguration — gets the safe default of no test users.

## 2. `feat(security)`: per-user avatar disk quota (default 50 MB)

Multer's 5 MB-per-file cap stops a single huge upload but does nothing about many small files accumulating. A user could create N packs and upload tens of thousands of small avatars, eventually pushing the shared disk to capacity.

Adds a cumulative-bytes gate before each upload:

- `getUserAvatarsUsage(userId)` walks the user's avatar directory recursively (each pack has its own subdir with thumbnail + original files) and sums file sizes. Tolerates ENOENT and stat/readdir races.
- Upload handler refuses with `413` if `currentUsage + req.file.size > quota`. Uses incoming buffer size as upper bound for the post-resize footprint — sharp generally shrinks, so this errs toward rejecting borderline uploads rather than letting the user creep past the cap.

Default quota: 50 MB, generous enough for 500+ post-resize avatars at sharp's `webp@quality:90` (typically 10–100 KB). Override via `AVATAR_USER_QUOTA_MB` env var.

**Not covered (deliberate scope limit; easy follow-ups if needed):**
- Pack `/clone` endpoint could still bypass by cloning a large pack to a new packId.
- No event logged on quota rejection (just the 413 response).

## 3. `chore`: add Node `engines` pin to backend/shared/frontend

Root `package.json` already declares `engines: { node: ">=22" }`, and `.nvmrc` pins to `22.14.0`. The three workspace packages had no `engines` field, so `npm install` in a workspace directly (rather than from root) skips the version check — easy to miss on first clone or in per-workspace CI installs. Adds the same constraint to each workspace.

## Scope / risk

Three commits, all backend-config-shaped. Backend `npm run build` passes each intermediate state. The test-user gate is a behavior change for anyone whose local dev relied on `NODE_ENV` being unset — they now need either `NODE_ENV=development` (the conventional value, which most dev setups already use) or the `ENABLE_TEST_USERS=true` opt-in.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: deprecated-claude-app/backend/src/routes/avatars.ts [hunk 1]
```diff
@@ -331,6 +380,25 @@ router.post('/packs/:packId/avatars', upload.single('avatar'), async (req: Multe
...
+    // creep past the cap.
+    const usage = await getUserAvatarsUsage(userId);
+    if (usage + req.file.size > AVATAR_USER_QUOTA_BYTES) {
+      const usedMB = (usage / (1024 * 1024)).toFixed(1);
+      const quotaMB = Math.round(AVATAR_USER_QUOTA_BYTES / (1024 * 1024));
+      return res.status(413).json({
+        error: 'Avatar storage quota exceeded',
+        details: `Using ${usedMB} MB of ${quotaMB} MB; this upload would exceed the limit. Delete unused avatars or contact an admin to raise the cap.`,
+      });
+    }
```
[the remark this answers is not in the document]
Comment (Meganeuridae) [reply deprecated-claude-app/backend/src/routes/avatars.ts]: Real race, confirmed. Fixed in 190943d with a per-user logical lock (`withUserUploadLock` — chain of manually-resolved promises keyed by userId). Synchronous register-as-new-tail step is atomic relative to other calls since there are no awaits between read/create/set, so the chain is well-formed. Smoke-tested in isolation: same-user serializes (A finishes before B starts), different users run in parallel, error in fn still releases the lock for the next caller, map size cleans back to 0. In-process only — multi-process scaling wouldn't coordinate, but that matches the rest of the write-path posture (P1-7); revisit if/when horizontal scale lands.

File: deprecated-claude-app/backend/src/routes/avatars.ts [hunk 2]
```diff
@@ -36,7 +36,56 @@ const AVATARS_BASE_PATH = getAvatarsBasePath();
...
+// Per-user disk quota for uploaded avatars. 50 MB is generous —
+// avatars after sharp's webp@quality:90 land at 10–100 KB typically,
+// so this allows 500+ avatars per user. Override with AVATAR_USER_QUOTA_MB
+// for ops that want to tune it.
+const AVATAR_USER_QUOTA_BYTES =
+  (Number(process.env.AVATAR_USER_QUOTA_MB) > 0
+    ? Number(process.env.AVATAR_USER_QUOTA_MB)
+    : 50) *
+  1024 *
+  1024;
```
[the remark this answers is not in the document]
Comment (Meganeuridae) [reply deprecated-claude-app/backend/src/routes/avatars.ts]: Good catch on the operator-surprise footgun. Fixed in 190943d using the structure from your suggestion: warn (with the offending value and a pointer to nginx for hard disable) when AVATAR_USER_QUOTA_MB is set but not a positive number, then fall through to the 50 MB default.
